shared/src/helpers/mathjax.js

```diff
@@ -0,0 +1,153 @@
+
+
+import 'shared/resources/styles/mathjax.less';
+
+const MATH_MARKER_BLOCK  = '\u200c\u200c\u200c'; // zero-width non-joiner
+const MATH_MARKER_INLINE = '\u200b\u200b\u200b'; // zero-width space
+
+const MATH_RENDERED_CLASS = 'math-rendered';
+const MATH_DATA_SELECTOR = `[data-math]:not(.${MATH_RENDERED_CLASS})`;
+const MATH_ML_SELECTOR   = `math:not(.${MATH_RENDERED_CLASS})`;
+const COMBINED_MATH_SELECTOR = `${MATH_DATA_SELECTOR}, ${MATH_ML_SELECTOR}`;
+const MATHJAX_CONFIG = {
+  showProcessingMessages: false,
+  tex2jax: {
+    displayMath: [[MATH_MARKER_BLOCK, MATH_MARKER_BLOCK]],
+    inlineMath:  [[MATH_MARKER_INLINE, MATH_MARKER_INLINE]],
+  },
+  'HTML-CSS': {
+    preferredFont: 'Tex',
+  },
+  availableFonts: [ 'Tex'],
+  styles: {
+    '#MathJax_Message': {
+      visibility: 'hidden', left: '', right: 0,
+    },
+    '#MathJax_MSIE_Frame': {
+      visibility: 'hidden', left: '', right: 0,
+    },
+  },
+};
+
+// Search document for math and [data-math] elements and then typeset them
+let typesetDocument = function(windowImpl) {
+  const latexNodes = [];
+  for (var node of Array.from(windowImpl.document.querySelectorAll(MATH_DATA_SELECTOR))) {
+    const formula = node.getAttribute('data-math');
+    // divs should be rendered as a block, others inline
+    if (node.tagName.toLowerCase() === 'div') {
+      node.textContent = `${MATH_MARKER_BLOCK}${formula}${MATH_MARKER_BLOCK}`;
+    } else {
+      node.textContent = `${MATH_MARKER_INLINE}${formula}${MATH_MARKER_INLINE}`;
+    }
+    latexNodes.push(node);
+  }
+
+  if (!isEmpty(latexNodes)) {
+    windowImpl.MathJax.Hub.Typeset(latexNodes);
+  }
+
+  const mathMLNodes = toArray(windowImpl.document.querySelectorAll(MATH_ML_SELECTOR));
+  if (!isEmpty(mathMLNodes)) {
+    // style the entire document because mathjax is unable to style individual math elements
+    windowImpl.MathJax.Hub.Typeset( windowImpl.document );
+  }
+
+  return windowImpl.MathJax.Hub.Queue(() =>
+    // Queue a call to mark the found nodes as rendered so are ignored if typesetting is called repeatedly
+    // uses className += instead of classList because IE
+    (() => {
+      const result = [];
+      for (node of Array.from(latexNodes.concat(mathMLNodes))) {
+        result.push(node.className += ` ${MATH_RENDERED_CLASS}`);
+      }
+      return result;
+    })()
+  );
+};
+
+// Install a debounce around typesetting function so that it will only run once
+// every Xms even if called multiple times in that period
+typesetDocument = debounce( typesetDocument, 100);
+
+// typesetMath is the main exported function.
+// It's called by components like HTML after they're rendered
+const typesetMath = function(root, windowImpl) {
+  // schedule a Mathjax pass if there is at least one [data-math] or <math> element present
+
+  if (windowImpl == null) { windowImpl = window; }
+
+  if ((__guard__(__guard__(windowImpl.MathJax, x1 => x1.Hub), x => x.Queue) != null) && root.querySelector(COMBINED_MATH_SELECTOR)) {
```

Comment:
hehe, yep the __guard__ is a big giveaway. I usually replace it with lodash's `get` but didn't bother since I think this approach is a dead-end.